engine/cld/config/network/config.go

```diff
@@ -172,17 +198,17 @@ func Load(filePaths []string, opts ...LoadOption) (*Config, error) {
 // LoadOption defines a function that modifies the load configuration.
 type LoadOption func(*loadConfig)
 
-// URLTransformer is a function that transforms the URLs in the Config.
-type URLTransformer func(*Config)
-
 // loadConfig holds the configuration for loading the config.
 type loadConfig struct {
-	URLTransformer URLTransformer
+	RPCURLTransformer URLTransformer
 }
 
-// WithURLTransform allows setting a custom URL transformer for the config.
-func WithURLTransform(t URLTransformer) func(opts *loadConfig) {
+// URLTransformer is a function that transforms a URL.
+type URLTransformer func(string) string
+
+// TransformRPCURLs is a load option that transforms the RPC URLs of the networks in the config.
+func TransformRPCURLs(t URLTransformer) LoadOption {
```

Comment:
Is this more readable and clearer?
```suggestion
func WithURLTransformer(t URLTransformer) LoadOption {
```